### DISCUSSION TITLE
Error when setup new project on Foundry

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#2904] [Q&A]
hello,
i m following the course, where it says to setup for the "FundMe" project on Foundry; but when i initialize with "forge init" everything is okay, when i move the FundMe and PriceConverter,everything is okay, when i install  the  "chainlink-brownie-contracts/contracts/" with "forge install https://github.com/smartcontractkit/chainlink-brownie-contracts@0.8.0 -no-commit",okay too; but when i make "forge build"; it gives me this error:

'''
2024-11-22T06:34:58.172830Z ERROR foundry_compilers_artifacts_solc::sources: error="/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-fund-me-f24/lib/chainlink-brownie-contracts/contracts": Is a directory (os error 21)
Error: failed to resolve file: "/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-fund-me-f24/lib/chainlink-brownie-contracts/contracts": Is a directory (os error 21); check configured remappings
        --> /mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-fund-me-f24/src/PriceConverter.sol
        lib/chainlink-brownie-contracts/contracts/
'''

where is the mistake i made?

**Comment by EngrPips** [comment]
This path `/mnt/c/Users/ahmed/OneDrive/Desktop/foundry_projects/foundry-f24/foundry-fund-me-f24/lib/chainlink-brownie-contracts/contracts`  shows that you are trying to import a folder and not a file, Can you check where you are using the path in your codebase and correct it?

**Comment by shoaib-eth** [comment]
update your `foundry.toml` to include the following in the `remappings`.

```
remappings = [
  '@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/',
]
```

I think you are importing `AggregatorV3Interface` for `priceFeed`.
So import `AggregatorV3Interface` like this 👇👇👇 (Copy this and paste in your code file)

```
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
```

then rebuild the project 

```
forge clean
```
and finally

```
forge build
```

**Reply by EngrPips** [reply]
Great heads-up

**Reply by ArnoldOrog** [reply]
I'm in tears rn. Thank you so much I thought my code will never work

**Comment by arbdev95** [comment]
done,resolved the problem,i forgot to change the path,was missing "/shared",but thank you guys for the answers!

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
